pallets/storage-proofs/src/lib.rs

```diff
@@ -0,0 +1,493 @@
+
+/// Edit this file to define custom logic or remove it if it is not needed.
+/// Learn more about FRAME and the core library of Substrate FRAME pallets:
+/// <https://docs.substrate.io/v3/runtime/frame>
+pub use pallet::*;
+
+// #[cfg(test)]
+// mod mock;
+
+// #[cfg(test)]
+// mod tests;
+
+// #[cfg(feature = "runtime-benchmarks")]
+// mod benchmarking;
+pub mod types;
+pub mod utils;
+
+use codec::FullCodec;
+use frame_support::{
+    inherent::IsFatalError,
+    pallet_prelude::*,
+    sp_runtime::{
+        traits::{AtLeast32BitUnsigned, CheckEqual, MaybeDisplay, SimpleBitOps},
+        RuntimeString,
+    },
+    traits::fungible,
+};
+use scale_info::prelude::fmt::Debug;
+use sp_trie::CompactProof;
+
+// TODO: Define this.
+const INHERENT_IDENTIFIER: InherentIdentifier = *b"todo____";
+
+#[frame_support::pallet]
+pub mod pallet {
+    use codec::FullCodec;
+    use frame_support::{
+        dispatch::DispatchResultWithPostInfo,
+        pallet_prelude::{ValueQuery, *},
+        sp_runtime::traits::{CheckEqual, MaybeDisplay, SimpleBitOps},
+        traits::fungible,
+    };
+    use frame_system::pallet_prelude::*;
+    use scale_info::prelude::fmt::Debug;
+    use sp_trie::CompactProof;
+    use types::SpFor;
+
+    use crate::types::*;
+    use crate::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Because this pallet emits events, it depends on the runtime's definition of an event.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The Storage Providers pallet.
+        /// To check if whoever submits a proof is a registered Storage Provider.
+        type StorageProviders: crate::StorageProvidersInterface<AccountId = Self::AccountId>;
+
+        /// The File System pallet.
+        /// To check if the root submitted in a proof is valid for the Storage Provider.
+        type FileSystem: crate::FileSystemInterface<
+            StorageProvider = SpFor<Self>,
+            ForestRoot = ForestRootFor<Self>,
+        >;
+
+        /// Type to access the Balances Pallet.
+        type NativeBalance: fungible::Inspect<Self::AccountId>
+            + fungible::hold::Inspect<Self::AccountId>
+            + fungible::freeze::Inspect<Self::AccountId>;
+
+        /// The type for keys that identify a file within a Merkle Patricia Forest.
+        /// Generally a hash (the output of a Hasher).
+        type FileKey: Parameter
+            + Member
+            + MaybeSerializeDeserialize
+            + Debug
+            + MaybeDisplay
+            + SimpleBitOps
+            + Ord
+            + Default
+            + Copy
+            + CheckEqual
+            + AsRef<[u8]>
+            + AsMut<[u8]>
+            + MaxEncodedLen
+            + FullCodec;
+
+        /// The type for a root of a Merkle Patricia Forest.
+        /// Generally a hash (the output of a Hasher).
+        type ForestRoot: Parameter
```

Comment:
have this as a type alias instead of a pallet config type. avoids any chance of having two different hashing types for leaves and root of the merkle trie.

Reply:
Solved in [fix: 🩹 Unifying FileKey and ForestRoot types into MerkleHash](https://github.com/Moonsong-Labs/storage-hub/pull/8/commits/ff3606697fa64d7b2e7753a5d74697eb3291e9f9)